### ISSUE TITLE
Feature Request - Timestamps as parameter

### ISSUE THREAD

**Issue by jdell64** [MISP/MISP#1703]
I am currently making a POST request to:
    
    https://<my-server-name>/events/restSearch/download
    Body: {"request": {"from":"2016-01-02", "to": "2016-01-03"}}

It would be great if I could be more granular with my time requests. Either a timestamp formatted:

    YYYY/mm/DDTHH:MM:SS

or just an epoch timestamp.

This alone would be pretty awesome, but it would be cool if I could also do this as a query parameter:

    https://<my-server-name>/events/restSearch/download?from=1480519222&to=1480539222

**Comment by iglocska** [comment]
Good point, we currently don't have any way of doing that. From and To rely on the event "date" field that is YYYY-MM-DD. 

The restsearch API has a "last" parameter that can be used alternatively, though it only allows a lower bound timestamp and it's based on the publish timestamp. The idea behind that was that you should be able to set up a recurring API call that will always fetch data from the past x amount of days(d)/hours(h)/minutes(m)/seconds(s).

So basically you can use

{"request": {"last"24h"}} to fetch all events published (or republished) in the past 24 hours.

**Comment by jdell64** [comment]
My only concern is that say I set up that reoccurring job to go for the past 1 hr, if my service goes down for for a bit, I may miss some jobs.

Now, if I keep track of the last time I queried, then I could adjust my window a bit, but I have no guarantee that I won't get duplicates.

**Comment by iglocska** [comment]
Very good point. I'll look into adding both options.

**Comment by jdell64** [comment]
Awesome! let me know if there is some way I can help.

**Comment by iglocska** [comment]
Pushed a fix, let me know if it doesn't do what you expect it to ;)

Basically POSTing
```
{"publish_timestamp":"1479827064"}
```
will return all events published after the given timestamp, posting
```
{"publish_timestamp":["1479827064", "1479830000"]}
```

will return all events published between those two timestamps.

For last edits use the "timestamp" parameters the same way. It's currently only there for restsearch and still needs to be documented.

**Comment by jdell64** [comment]
Not sure about php, but I know in many languages the array doesn't guarantee order of array. Does

    {"publish_timestamp":["1479827064", "1479830000"]}

yield the same results as 

    {"publish_timestamp":["1479830000", "1479827064"]}   // swapped the numbers

? And thanks for the quick turnaround!

**Comment by iglocska** [comment]
in php it guarantees it. Basically PHP always uses key->value style arrays internally for lists (assigning it incrementing indeces left to right), on json decoding it will automatically convert it to:

```
array(0 => "1479827064", 1 => "1479830000")
```

As a fun side-story, you get some funky results if you remove an element from a list and JSON encode it after.

```
$test = array('a', 'b', 'c');
echo json_encode($test);
````
this will convert it to:
```
["a", "b", "c"]
````
However if we do the following after:
```
unset($test[1]);
echo json_encode($test);
```
PHP will be missing index 1, and it will think it's no longer a list so the result will be:

```
{"0": "a", "2": "c"}
```

tl;dr:
First number will always be lower bound, second number upper bound.
